pkg/validatorfactory/factory.go

```diff
@@ -43,12 +52,12 @@ func (v *ValidatorEntry) IsNamespaceScoped() bool {
 	return v.namespaceScoped
 }
 
-func (v *ValidatorEntry) SchemaValidator() *validate.SchemaValidator {
+func (v *ValidatorEntry) SchemaValidator() validation.SchemaValidator {
 	if v.schemaValidator != nil {
 		return v.schemaValidator
 	}
 
-	v.schemaValidator = validate.NewSchemaValidator(v.Schema, nil, "", strfmt.Default)
+	v.schemaValidator = &BasicValidatorAdapter{SchemaValidator: validate.NewSchemaValidator(v.Schema, nil, "", strfmt.Default)}
```

Comment:
@apelisse @alexzielenski do we need to create a validator adapter ?

cc @fjogeleit

Reply:
The interface of the Validator changed in https://pkg.go.dev/k8s.io/apiextensions-apiserver@v0.28.2/pkg/apiserver/validation#SchemaValidator

They do something similar:

https://github.com/kubernetes/apiextensions-apiserver/blob/v0.28.2/pkg/apiserver/validation/validation.go#L46